Client/MirScenes/Dialogs/NPCDialogs.cs

```diff
@@ -453,6 +567,448 @@ private void NewColour(string text, string colour, Point p)
+        {
+            if (int.TryParse(monsterIdx, out int idx))
+            {
+                var monster = GameScene.MonsterInfoList.FirstOrDefault(x => x.Index == idx);
+                if (monster != null)
+                {
+                    return string.Join("\n", new[]
+                    {
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipLevel, monster.Level),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipHealth, monster.Stats[Stat.HP]),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipExperience, monster.Experience),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipDC, monster.Stats[Stat.MinDC], monster.Stats[Stat.MaxDC]),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipMC, monster.Stats[Stat.MinMC], monster.Stats[Stat.MaxMC]),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipAC, monster.Stats[Stat.MinAC], monster.Stats[Stat.MaxAC]),
+                        GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterTooltipMAC, monster.Stats[Stat.MinMAC], monster.Stats[Stat.MaxMAC])
+                    });
+                }
+                GameScene.RequestMonsterInfo(idx);
+                return GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.MonsterIndexLoading, idx);
+            }
+            return GameLanguage.ClientTextMap.GetLocalization(ClientTextKeys.InvalidMonsterIndex);
+        }
+
+        private string GetNPCInfo(string npcIdx)
+        {
+            if (int.TryParse(npcIdx, out int idx))
+            {
+                var npc = GameScene.NPCInfoList.FirstOrDefault(x => x.Index == idx);
+                if (npc != null)
+                {
+                    string mapName = string.Empty;
+
+                    // Try to get from current map if we're on that map
+                    if (GameScene.Scene.MapControl != null && GameScene.Scene.MapControl.Index == npc.MapIndex)
+                    {
+                        mapName = GameScene.Scene.MapControl.Title;
+                    }
+                    // Otherwise try MapInfoList
+                    else if (npc.MapIndex > 0 && GameScene.MapInfoList.TryGetValue(npc.MapIndex, out var mapInfo))
+                    {
+                        if (mapInfo?.MapInfo != null && !string.IsNullOrEmpty(mapInfo.MapInfo.Title))
+                        {
+                            mapName = mapInfo.MapInfo.Title;
+                        }
+                    }
+                    // Last resort: try searching all map info records
+                    else if (npc.MapIndex > 0)
+                    {
+                        var foundMap = GameScene.MapInfoList.Values.FirstOrDefault(x => x.MapInfo?.BigMap == npc.MapIndex);
```

Comment:
Why is it searching big map? Plus we've just done the search. If that didn't work this one won't either.